pkg/controllers/customcluster_controller.go

```diff
@@ -47,16 +76,515 @@ type CustomClusterController struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *CustomClusterController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := ctrl.LoggerFrom(ctx)
+
+	// Fetch the customCluster instance.
+	customCluster := &v1alpha1.CustomCluster{}
+	if err := r.Client.Get(ctx, req.NamespacedName, customCluster); err != nil {
+		return ctrl.Result{RequeueAfter: RequeueAfter}, err
+	}
+
+	log = log.WithValues("customCluster", klog.KObj(customCluster))
+	ctx = ctrl.LoggerInto(ctx, log)
+
+	phase := customCluster.Status.Phase
+
+	if len(phase) == 0 {
+		customCluster.Status.Phase = v1alpha1.PendingPhase
+		log.Info("customCluster's phase changes from nil to Pending")
+		if err := r.Status().Update(ctx, customCluster); err != nil {
+			log.Error(err, "phase can not update")
+			return ctrl.Result{RequeueAfter: RequeueAfter}, err
+		}
+	}
+
+	if phase == v1alpha1.FailedPhase {
+		return ctrl.Result{Requeue: false}, nil
```

Comment:
This will fall into dead loop if failed phase set.  It has no chance to set to another phase